### PR TITLE
move global data and node selector out of the bpfcommon data struct

### PR DESCRIPTION
**Pull request by msherif1234**
preq before defining new application CRD

### CODE REVIEW

File: apis/v1alpha1/kprobeProgram_types.go [hunk 1]
```diff
@@ -50,6 +50,23 @@ type KprobeProgram struct {
 // +kubebuilder:printcolumn:name="RetProbe",type=boolean,JSONPath=`.spec.retprobe`
 // +kubebuilder:validation:XValidation:message="offset cannot be set for kretprobes",rule="self.retprobe == false || self.offset == 0"
 type KprobeProgramSpec struct {
+	KprobeProgramInfo `json:",inline"`
+
+	// NodeSelector allows the user to specify which nodes to deploy the
+	// bpf program to.  This field must be specified, to select all nodes
+	// use standard metav1.LabelSelector semantics and make it empty.
+	NodeSelector metav1.LabelSelector `json:"nodeselector"`
```
[the remark this answers is not in the document]
Comment (msherif1234) [reply apis/v1alpha1/kprobeProgram_types.go]: done

File: bundle/manifests/bpfman.io_fentryprograms.yaml [hunk 2]
```diff
@@ -149,11 +149,13 @@ spec:
                           items:
                             type: string
                           type: array
+                          x-kubernetes-list-type: atomic
```
Comment (anfredette) [inline bundle/manifests/bpfman.io_fentryprograms.yaml:152]: Did you do anything to cause this to change, or did it just happen when you regenerated the crds?

Comment (anfredette) [reply bundle/manifests/bpfman.io_fentryprograms.yaml:152]: Possibly due to an update in the tools.  I think that's the default anyway.

**Review by anfredette — APPROVED** [review]
LGTM

**Pull request merged**